internal/config/yaml_parser_test.go

```diff
@@ -16,253 +16,234 @@ package config
 	_, err := ParseConfigFile("testdata/metadata_cache_config_invalid_stat-cache-max-size-mb.yaml")
 
-	AssertNe(nil, err)
-	AssertThat(err, oglematchers.Error(oglematchers.HasSubstr(StatCacheMaxSizeMBInvalidValueError)))
+	assert.ErrorContains(t.T(), err, StatCacheMaxSizeMBInvalidValueError)
 }
 
 func (t *YamlParserTest) TestReadConfigFile_MetatadaCacheConfig_StatCacheSizeNotSet() {
 	mountConfig, err := ParseConfigFile("testdata/metadata_cache_config_stat-cache-max-size-mb_unset.yaml")
 
-	AssertEq(nil, err)
-	AssertNe(nil, mountConfig)
-	AssertEq(StatCacheMaxSizeMBUnsetSentinel, mountConfig.MetadataCacheConfig.StatCacheMaxSizeMB)
+	assert.NoError(t.T(), err)
+	assert.NotNil(t.T(), mountConfig)
+	assert.Equal(t.T(), StatCacheMaxSizeMBUnsetSentinel, mountConfig.MetadataCacheConfig.StatCacheMaxSizeMB)
 }
 
 func (t *YamlParserTest) TestReadConfigFile_MetatadaCacheConfig_StatCacheSizeTooHigh() {
 	_, err := ParseConfigFile("testdata/metadata_cache_config_stat-cache-max-size-mb_too_high.yaml")
 
-	AssertNe(nil, err)
-	AssertThat(err, oglematchers.Error(oglematchers.HasSubstr(StatCacheMaxSizeMBTooHighError)))
+	assert.ErrorContains(t.T(), err, StatCacheMaxSizeMBTooHighError)
 }
 
 func (t *YamlParserTest) TestReadConfigFile_GrpcClientConfig_invalidConnPoolSize() {
 	_, err := ParseConfigFile("testdata/grpc_client_config/invalid_conn_pool_size.yaml")
 
-	AssertNe(nil, err)
-	AssertTrue(strings.Contains(err.Error(), "error parsing grpc-config: the value of conn-pool-size can't be less than 1"))
+	assert.ErrorContains(t.T(), err, "error parsing grpc-config: the value of conn-pool-size can't be less than 1")
 }
 
 func (t *YamlParserTest) TestReadConfigFile_GrpcClientConfig_unsetConnPoolSize() {
 	mountConfig, err := ParseConfigFile("testdata/grpc_client_config/unset_conn_pool_size.yaml")
 
-	AssertEq(nil, err)
-	AssertNe(nil, mountConfig)
-	AssertEq(DefaultGrpcConnPoolSize, mountConfig.GrpcClientConfig.ConnPoolSize)
+	assert.NoError(t.T(), err)
+	assert.NotNil(t.T(), mountConfig)
+	assert.Equal(t.T(), DefaultGrpcConnPoolSize, mountConfig.GrpcClientConfig.ConnPoolSize)
 }
 
 func (t *YamlParserTest) TestReadConfigFile_FileSystemConfig_InvalidIgnoreInterruptsValue() {
 	_, err := ParseConfigFile("testdata/file_system_config/invalid_ignore_interrupts.yaml")
 
-	AssertNe(nil, err)
-	AssertTrue(strings.Contains(err.Error(), "error parsing config file: yaml: unmarshal errors:\n  line 2: cannot unmarshal !!str `abc` into bool"))
+	assert.ErrorContains(t.T(), err, "error parsing config file: yaml: unmarshal errors:\n  line 2: cannot unmarshal !!str `abc` into bool")
 }
 
 func (t *YamlParserTest) TestReadConfigFile_FileSystemConfig_UnsetIgnoreInterruptsValue() {
 	mountConfig, err := ParseConfigFile("testdata/file_system_config/unset_ignore_interrupts.yaml")
 
-	AssertEq(nil, err)
-	AssertNe(nil, mountConfig)
-	AssertEq(false, mountConfig.FileSystemConfig.IgnoreInterrupts)
+	assert.NoError(t.T(), err)
```

Comment:
Not sure but other places I used assert.Nil. Should we maintain consistency ?

Reply:
I think both are correct only but assert.NotError is better because with assert.Nil our intention is to check it's not an error. So, you can keep the assert.Nil but going forward you can use assert.NotError.